src/InlineEntityForm/EntityInlineEntityFormHandler.php

```diff
@@ -270,4 +273,39 @@ protected function buildChildFormState(EntityFormInterface $controller, FormStat
     return $child_form_state;
   }
 
+  /**
+   * Cleans up the form state for a submitted entity form.
+   *
+   * After field_attach_submit() has run and the form has been closed, the form
+   * state still contains field data in $form_state['field']. Unless that
+   * data is removed, the next form with the same #parents (reopened add form,
+   * for example) will contain data (i.e. uploaded files) from the previous form.
+   *
+   * @param $entity_form
+   *   The entity form.
+   * @param $form_state
+   *   The form state of the parent form.
+   */
+  public static function submitCleanFormState(&$entity_form, FormStateInterface $form_state) {
+    $info = \Drupal::entityManager()->getDefinition($entity_form['#entity_type']);
+    if (!$info->get('field_ui_base_route')) {
+      // The entity type is not fieldable, nothing to cleanup.
+      return;
+    }
+
+    $bundle = $entity_form['#entity']->bundle();
+    $instances = \Drupal::entityManager()->getFieldDefinitions($entity_form['#entity_type'], $bundle);
+    foreach ($instances as $instance) {
+      $field_name = $instance->getName();
+      if (!empty($entity_form[$field_name]['#parents'])) {
+        $parents = $entity_form[$field_name]['#parents'];
+        array_pop($parents);
+        if (is_array($parents)) {
```

Comment:
This seems strange with the above as array_pop needs an array? Should this really be !empty($parents)